### PR TITLE
Add Google Analytics

### PR DESCRIPTION
**Pull request by smarzban**

### CODE REVIEW

**Comment by smarzban** [comment]
@smn @bruskiza @miltontony ready for review

File: ndohyep/settings/base.py [hunk 1]
```diff
@@ -216,3 +218,22 @@
...
+
+# Middleware + Celery
+
+MIDDLEWARE_CLASSES = (
+    'google_analytics.middleware.GoogleAnalyticsMiddleware',
+)
+
+CELERY_IMPORTS = ('google_analytics.tasks')
+CELERY_RESULT_BACKEND = 'djcelery.backends.database:DatabaseBackend',
+CELERY_RESULT_BACKEND = 'djcelery.backends.cache:CacheBackend',
```
Comment (miltontony) [inline ndohyep/settings/base.py]: You don't need to specify `CELERY_RESULT_BACKEND`

**Comment by miltontony** [comment]
I've never needed `ndohyep/celery.py`

**Comment by miltontony** [comment]
It's usually enough to add this in `settings.py`

``` python
import djcelery
djcelery.setup_loader()
```

**Comment by smarzban** [comment]
@miltontony @smn @bruskiza ready for review

File: ndohyep/settings/base.py [hunk 2]
```diff
@@ -216,3 +220,21 @@
...
     os.path.join(PROJECT_ROOT, 'ndohyep'),
 )
+
+
+# Google Analytics
+
+GOOGLE_ANALYTICS = {
+    'google_analytics_id': '',
+}
+GOOGLE_ANALYTICS_IGNORE_PATH = ['', ]
```
Comment (smn) [inline ndohyep/settings/base.py]: Let's make sure we add `'/health/'` here and the `google_analytics_id` is something we set at deploy time with puppet.

File: ndohyep/settings/base.py [hunk 3]
```diff
@@ -216,3 +220,21 @@
...
+
+
+# Middleware + Celery
+
+MIDDLEWARE_CLASSES = (
+    'google_analytics.middleware.GoogleAnalyticsMiddleware',
+)
+
+
+CELERY_IMPORTS = ('google_analytics.tasks')
```
Comment (smn) [inline ndohyep/settings/base.py]: This needs to be a tuple: `('google_analytics.tasks',)` (note the extra comma).

File: ndohyep/urls.py [hunk 4]
```diff
@@ -23,6 +23,7 @@
     url(r'commenting/reply/(?P<parent_id>\d+)/$', CommentReplyForm.as_view(),
         name='comments-reply'),
     url(r'', include(wagtail_urls)),
+    url(r'^djga/', include('google_analytics.urls')),
```
Comment (smn) [inline ndohyep/urls.py:25]: I think we still need to reference the image that does the analytics tracking from the templates somewhere.

Comment (miltontony) [inline ndohyep/urls.py:25]: @smn not relevant if we're using the celery middleware

**Comment by smarzban** [comment]
@smn @miltontony how can I fix the travis error? it says `No module named urls`

File: ndohyep/settings/base.py [hunk 5]
```diff
@@ -216,3 +220,21 @@
...
+GOOGLE_ANALYTICS = {
+    'google_analytics_id': '',
+}
+GOOGLE_ANALYTICS_IGNORE_PATH = ['/health/', ]
+
+
+# Middleware + Celery
+
+MIDDLEWARE_CLASSES = (
+    'google_analytics.middleware.GoogleAnalyticsMiddleware',
```
Comment (miltontony) [inline ndohyep/settings/base.py]: This is wrong. You need to add it to other `MIDDLEWARE_CLASSES` on line 100

**Comment by smarzban** [comment]
@miltontony still travis is failing :(

File: requirements.txt [hunk 6]
```diff
@@ -1,2 +1,4 @@
 molo.core==2.4.1
 molo.commenting<1.0.0,>=0.2.3
+django-google-analytics
```
Comment (miltontony) [inline requirements.txt]: Found the problem. This should be `django-google-analytics-app`

Comment (smarzban) [inline requirements.txt]: yaaay thanks @miltontony

**Comment by smarzban** [comment]
@miltontony what causes this error?  `[Errno 111] Connection refused`

**Comment by miltontony** [comment]
:+1:

**Pull request merged**